### PR TITLE
Us 129074

### PR DESCRIPTION
**Pull request by osalamakhagl**
Added ability to filter apps version in the our store.

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/smart/appfiltering/data/Age.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+
+package cm.aptoide.pt.smart.appfiltering.data;
```
Comment (osalamakhagl) [inline app/src/main/java/cm/aptoide/pt/smart/appfiltering/data/Age.java:2]: cm.aptoide.pt.smart.appfiltering.data This package only contains POJO objects for json response.

File: app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private void rebuildLatestVersionIfNeeded(GetAppResponse response, List<AppToRemove> filterList) {
+        boolean shouldRebuild = shouldRebuildLatestVersion(getLatestVersion(response), filterList);
+        if (shouldRebuild) {
+            List<Version> versions = response.getNodes().getVersions().getList();
```
Comment (shansen2021) [inline app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java:56]: Is it possible for anything in this chain to be null?

File: app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                    }
+                }
+                rebuildLatestVersion(response, maxVersion);
+            }
+        }
+    }
+
+    private void rebuildLatestVersion(GetAppResponse response, Version version) {
+        String apkLink = buildLinkForTheSpecificVersion(response, version);
+        File apkFile = response.getNodes().getMeta().getData().getFile();
```
Comment (shansen2021) [inline app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java:71]: Is it possible for anything in this chain to be null?

File: app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        apkFile.setVername(version.getFile().getVername());
+        apkFile.setVercode(version.getFile().getVercode());
+        apkFile.setPath(apkLink);
+        apkFile.setMd5sum(version.getFile().getMd5sum());
+    }
+
+
+
+    private String getLatestVersion(GetAppResponse response) {
+        return response.getNodes().getMeta().getData().getFile().getVername();
```
Comment (shansen2021) [inline app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java:82]: And one more time - is it possible for anything in this chain to be null?

File: app/src/main/java/cm/aptoide/pt/smart/appfiltering/ApplicationRemoveWrapper.java [hunk 5]
```diff
@@ -7,34 +7,58 @@
...
+            JSONObject object = filterHelper.filterVersionRequest(bodyString, FilteringList.filteringList);
+            if (object != null) {
+                return rebuildResponse(originalResponse, object, chain);
             }
         }
         return originalResponse;
     }
+
+
+    private Response rebuildResponse(Response originalResponse, JSONObject modifiedResposne, Interceptor.Chain chain) {
```
Comment (shansen2021) [inline app/src/main/java/cm/aptoide/pt/smart/appfiltering/ApplicationRemoveWrapper.java]: Typo in "modifiedResponse"

File: app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java (hunk 2 above)
Comment (pein95) [reply app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java:56]: I've added nullcheck method before this function is called.

File: app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java (hunk 3 above)
Comment (pein95) [reply app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java:71]: Added null check to the filterResponse method.

File: app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java (hunk 4 above)
Comment (pein95) [reply app/src/main/java/cm/aptoide/pt/smart/appfiltering/FilterHelper.java:82]: Added null check to the filterResponse method.

File: app/src/main/java/cm/aptoide/pt/smart/appfiltering/ApplicationRemoveWrapper.java (hunk 5 above)
Comment (pein95) [reply app/src/main/java/cm/aptoide/pt/smart/appfiltering/ApplicationRemoveWrapper.java]: Fixed

**Review by shansen2021 — APPROVED** [review]

**Pull request merged**